api/src/routes/goals/Goals.ts

```diff
@@ -9,8 +10,30 @@ goals.get('/:id', (req, res) => {
 });
 
 goals.post('/:id', async (req, res) => {
+    if (!req.session.user) {
+        res.sendStatus(401);
+        return;
+    }
+
+    const game = await gameForGoal(req.params.id);
+
+    // If no game could be found, the goal probably doesn't exist
+    if(!game) {
+        res.sendStatus(404);
+        return;
+    }
+
+    const [owner, mod] = 
+        await Promise.all([isOwner(game.slug, req.session.user), isModerator(game.slug, req.session.user)]);
```

Comment:
Would it be better to update `isMod` to also check for owners? Would allow us to condense this into a single awaited promise rather than firing off two database checks, at the cost of making the mod query a little more complex. We'd have to make sure we don't rely on `isMod` to create mod lists, though I don't think we do (since a single query is way better for performance anyway). Anywhere a mod has permissions, an owner also does, so I don't think the distinction is strictly necessary